### PR TITLE
Expand development guide on pre-commit usage

### PR DESCRIPTION
**Pull request by StanczakDominik**
Follow up to #961 that expands the development guide a little, putting pre-commit up-front above the style guide.

I am actually going to go against the new review system and ask for review from each of you (if you can find the time), because:

* @namurphy you did #961 and I would like to see what you think of the follow-up changes!
* @rocco8773 I would prefer to know if this suggested set-up works with your Windows PyCharm setup before we merge this in!
* also @pheuer I promised I'd do this in https://github.com/PlasmaPy/PlasmaPy/pull/895#issuecomment-770159685 and I wanted to know if this sounds reasonable :)

- [x] I have added a changelog entry for this pull request.

- [x] If adding new functionality, I have added tests and
      docstrings.

- [x] I have fixed any newly failing tests.

---

EDIT:

I've gone ahead and prepared stuff for pre-commit.ci; it's looking straight up superior to our current solution.

### CODE REVIEW

**Comment by StanczakDominik** [comment]
I also just checked and https://pre-commit.ci/ actually got off the ground in the last few months. We could try that!

**Comment by StanczakDominik** [comment]
Well, I went ahead and tried pre-commit.ci on my fork; https://github.com/StanczakDominik/PlasmaPy/pull/5 is how it looks like. Kinda cool, no?

**Comment by StanczakDominik** [comment]
Given today's telecon and the intent to hold off on this a little, I'll revert to be926fe and we can discuss the wording on that version :)

File: docs/development/code_guide.rst [hunk 1]
```diff
@@ -13,6 +13,42 @@ later.
...
+
+PlasmaPy has a configuration for the `pre-commit framework
+<https://pre-commit.com/>`_ that takes care of style mostly automatically.
+Install it with `pip install pre-commit`, then use `pre-commit install` within
+the repository.
+
+This will cause pre-commit to download the right versions of linters we use,
+then run an automated style checking suite on every commit.  Do note that this
+works better with a `git add`, then `git commit` workflow than a `git commit
+-a` workflow - that way, you can check via `git diff` what the automated
```
Comment (namurphy) [inline docs/development/code_guide.rst:27]:
```suggestion
-a` workflow — that way, you can check via `git diff` what the automated
```
Unicode!  😬

File: docs/development/code_guide.rst [hunk 2]
```diff
@@ -13,6 +13,42 @@ later.
...
+It includes
+
+* `black <https://black.readthedocs.io/en/stable/>`_ to automatically
+  format code and ensure a consistent code style throughout the
+  package
+* `isort <https://pycqa.github.io/isort/>`_ to
+  automatically sort imports.
+* `nbqa <https://github.com/nbQA-dev/nbQA>`_ to automatically apply the above
+  to notebooks as well.
+* a few tools for requirements.txt, .yml files and the like.
```
Comment (namurphy) [inline docs/development/code_guide.rst:47]:
```suggestion
* a few tools for `requirements.txt`, `.yml` files and the like.
```

File: docs/development/code_guide.rst [hunk 3]
```diff
@@ -13,6 +13,42 @@ later.
...
+<https://github.com/PlasmaPy/PlasmaPy/blob/master/.pre-commit-config.yaml>`_.
+It includes
+
+* `black <https://black.readthedocs.io/en/stable/>`_ to automatically
+  format code and ensure a consistent code style throughout the
+  package
+* `isort <https://pycqa.github.io/isort/>`_ to
+  automatically sort imports.
+* `nbqa <https://github.com/nbQA-dev/nbQA>`_ to automatically apply the above
+  to notebooks as well.
```
Comment (namurphy) [inline docs/development/code_guide.rst:46]:
```suggestion
  to example notebooks as well.
```

**Review by namurphy — APPROVED** [review]
Looks good!  I only have a few minor stylistic suggestions.  Thank you for updating this!

**Pull request merged**